### DISCUSSION TITLE
Adding a value from a Details View on the page to a Grid Field in Edit mode

### DISCUSSION THREAD

**Discussion by djbbiz** [cortexrd/Knack-Toolkit-Library#200] [Q&A]
Hey guys, I have a question as I didn't actually find a good solution in the docs but maybe is possible.

I have an Account Details view on a page of a user (not the logged in user).

I have a Grid View for another Object on the same page that also has a connected Account Field and the fields are editable.

The Grid View is basically a Template. A few fields are filled in and the record is inserted as a new record into another Object using an Action button.

The action button also deletes the fields in the template that were just filled in after it inserts a copy into the second object and the Grid is ready to use again.

One of the fields that the user has to fill in the first Grid on the page is the connected Account from a dropdown.

The Account that needs to be selected is already in the Details View at the top of the page.

Is there a way to use the details view Account ID when the action is clicked to insert a new record in object 2? That way the user is less likely to make an error in selecting the correct connected account and doesn't have to keep doing it for every record they choose to copy over to object 2.

**Comment by cortexrd** [comment]
Hi Dean,

There is no method to do this currently using a keyword or other.
But the KTL code has all that is needed to do it.  I actually do this countless of places in my client's app.

Le me think of something...

How about **_val** for **Set Value** ?

It could be something like this:

`_val=[field_1], [ktlRefVal, field_2, view_2]`

It would read the value of field_2 in view_2 (a details view always), then put it in field_1 of the current view.
The field_1 type could be anything: short text, connection, date/time, etc.

Norm

**Reply by djbbiz** [reply]
That sounds like an elegant solution to a problem I have seen posted many times in the community

**Comment by cortexrd** [comment]
BTW, from your use case, you might be interested in the **_cpyfrom** keyword, as it does something very similar to what you're doing:  copying a record from a template to a new view.

I use this for my Survey pages.  The Survery Questons are the template, and I copy each one to the User's Answers, where he can fill in the blanks.

Norm

**Reply by djbbiz** [reply]
I looked at _cpyfrom but it only supports the logged in user and I need it to apply to an Account that is selected in the details view not the logged in user.

**Reply by cortexrd** [reply]
Indeed, the keyword does support the logged-in account, but it also support **any** connected field's value from a details view.  In my case, that was the Survey name.

So it can do both.

Norm

**Comment by djbbiz** [comment]
As further info, your use case worked because you are copying the entire table to the user where my table has over 800 records all essentially like questions for different clinical issues a doctor is evaluating for a patient. The Doctor may only need for 5-8 of those records to be copied over to the User's file and he has to decide that based on his examination and add context to that record one by one. So in my case the Doctor is working inside the template, he fills in his data for a specific record/disease condition, clicks an Action Link named Add and it copies that template record with the Doctor's added content over into another Table and assigns the user which has to be selected in each record right now. So I can't copy all 800 records each time as the volume would accumulate quickly

**Reply by cortexrd** [reply]
I perfectly agree with you in that case.

But still, I'm convinced that it would be possible to do this:

1. Use a Quick Toggle (_qt) on a Yes/No field to select the rows from the template records.
2. The doctor edits the text fields he needs
3. Applying a User Filter that only keeps those selected as Yes.
4. Do the _cpyfrom by clicking on the button with the label of his choice

Don't forget that you can set the _cpyfrom keyword parameters to copy only the fields you want, set the client's name from a details view, etc.

I think this would be a very nice solution.

Norm

**Comment by djbbiz** [comment]
I am still not sure that the feature set would work as I need to experiment more, as this one is a bit more complex, but unfortunately it won't work in my case due to the fact that all non-editable fields will be ignored. I have 4 fields they can edit but several fields that contain information and health codes for insurance that I need to copy as well. 

Your solution of _val=[field_1], [ktlRefVal, field_2, view_2] seems like a more straight forward way to perform the function as I need it. The value of field_1 in View_1 would get entered into the field_2 in view_2 for any record I copy when the action link is triggered.

In general there is a lot of potential use for setting a variable to a value in a field and using that value elsewhere on the page in a different view but my use case is one I have run into several times and have seen others trying to do this with JavaScript by creating checkboxes and a Copy Button for the checked records. 

In my case I need the doctor to be able to click an Action link or any button for just the one record each time as there is too much data and too many paginations of the table to risk losing the clinical and administrative data they are entering. I need them to focus on one disease condition, evaluate it, document their notes and images and click to save it to another table which is handled by the Action Link in my case and an "Insert a new record to a connected table".

**Comment by djbbiz** [comment]
I realized part of my solution wasn't scalable for multiple doctors working with the template simultaneously, so I made a change to use the Action Link to create the new Record and then edit it with a form and a hidden account field and some JavaScript to change the Account programmatically. Working well with some keyword codes to refresh the parent views of modal form to edit and a keyword to hide the Account Field on the form so they can't accidentally change it. Problem solved for now.
